pandas/io/parsers.py

```diff
@@ -2413,11 +2422,14 @@ def _infer_columns(self):
                 for i, c in enumerate(line):
                     if c == '':
                         if have_mi_columns:
-                            this_columns.append('Unnamed: %d_level_%d'
-                                                % (i, level))
+                            col_name = ("Unnamed: {i}_level_{level}"
+                                        .format(i=i, level=level))
                         else:
-                            this_columns.append('Unnamed: %d' % i)
+                            col_name = "Unnamed: {i}".format(i=i)
+
                         unnamed_count += 1
+                        unnamed_cols.add(col_name)
+                        this_columns.append(col_name)
```

Comment:
so we still need unnamed_count ?
does unnamed_cols need to be a set or can it be a list? (you can always convert to a set later if needed)